### PR TITLE
chore: final touches to ExperimentConfig V0

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

Checklist of work completed:

- Meticulously recompare each expconf struct against its tip-of-master version.

- Bring in updates to pkg/model configs that have landed since the expconf
work started.

- Tolerate resources.slots, to avoid unnecessary user breakages.  Removing
it can happen as part of ExperimentConfigV1.

- Add the mysteriously-missing environment.registry_auth.

- Fix a couple of bugs in the python schema test.

### CODE REVIEW

**Comment by stoksc** [comment]
@rb-determined-ai this is probably where https://github.com/determined-ai/determined/commit/9e3c69d62b0d45f1a39bc1ee1531b943679a4f2d needs to be snagged?

**Review by stoksc — COMMENTED** [review]
lgtm, save the conflicts I just caused 😖

File: harness/determined/common/schemas/expconf/_v0.py [hunk 1]
```diff
@@ -235,14 +235,38 @@ def from_dict(
         return super().from_dict(d, prevalidated)
 
 
+class RegistryAuthConfigV0(schemas.SchemaBase):
```
Comment (shiyuann) [inline harness/determined/common/schemas/expconf/_v0.py:238]: Is `_v0.py` also a generated file?

Comment (rb-determined-ai) [reply harness/determined/common/schemas/expconf/_v0.py:238]: No it's not.

File: schemas/test_cases/v0/misc.yaml [hunk 2]
```diff
@@ -119,3 +119,10 @@
     bucket: asdf
     bucket_directory_path: /asdf/asdf
     local_cache_container_path: /asdf/asdf
+
+- name: allow useless slots config in resources
+  matches:
+    - http://determined.ai/schemas/expconf/v0/resources.json
+  case:
+    slots: 1
```
Comment (stoksc) [inline schemas/test_cases/v0/misc.yaml:127]: 👍

**Review by stoksc — APPROVED** [review]

**Pull request merged**